src/main/java/seedu/address/ui/TaskCard.java

```diff
@@ -52,7 +52,13 @@ public TaskCard(Task task, int displayedIndex) {
         date.setText(task.getDate() != null ? task.getDate().toString() : "");
         Label progressLabel = new Label(task.getProgress().name().toLowerCase());
         progress.getChildren().add(progressLabel);
-        dueDate.setText(LocalDate.now().until(task.getDate(), DAYS) + " Days");
+
+        if (task.getDate() != null) {
+            dueDate.setText(LocalDate.now().until(task.getDate(), DAYS) + " Days");
+        } else {
+            dueDate.setText("");
```

Comment:
Might want to null date to "-", so its obvious on the UI that the duedate is not set, instead of a blank